components/plan-build/bin/environment.sh

```diff
@@ -0,0 +1,466 @@
+# Functions for resolving the runtime environment of a package.
+
+declare -A __runtime_environment
+declare -A __buildtime_environment
+
+declare -A __runtime_environment_provenance
+declare -A __buildtime_environment_provenance
+
+# Layer all direct dependencies' environment together.
+#
+# Priority is last-one-wins, based on the order in which you define
+# the dependencies in your plan file.
+#
+# (I'm sure there are many other common variables we could add here;
+# PRs welcome!)
+declare -A -g __well_known_aggregate_env_vars=(
```

Comment:
Maybe these belong in scaffolding instead of the core of plan build. Probably safe for a first pass to leave them here for now

Reply:
@reset interesting idea... could give us more flexibility in pushing updates independently of the rest of the build tooling, if nothing else.